tools/mo/openvino/tools/mo/moc_frontend/extractor.py

```diff
@@ -22,27 +22,63 @@ def decode_name_with_port(input_model: InputModel, node_name: str, framework="")
     found_nodes = []
     found_node_names = []
 
-    node = input_model.get_place_by_tensor_name(node_name)
-    if node:
-        found_node_names.append('Tensor:' + node_name)
-        found_nodes.append(node)
+    def try_get_tensor(node_name):
+        tensor = input_model.get_place_by_tensor_name(node_name)
+        if tensor:
+            return tensor
+        if node_name.count(":"):
+            regexp_pre = r'(\d+):(.+)'
+            match_pre = re.search(regexp_pre, node_name)
+            if match_pre:
+                port, name = node_name.split(":")
+                node = input_model.get_place_by_operation_name(name)
+                if node:
+                    tensor = node.get_source_tensor(input_port_index=int(port))
+                    if tensor:
+                        return tensor
+                tensor = input_model.get_place_by_tensor_name(name)
+                if tensor:
+                    return tensor
+            regexp_post = r'(.+):(\d+)'
+            match_post = re.search(regexp_post, node_name)
+            if match_post:
+                name, port = node_name.split(":")
+                node = input_model.get_place_by_operation_name(name)
+                if node:
+                    tensor = node.get_target_tensor(output_port_index=int(port))
+                    if tensor:
+                        return tensor
+                tensor = input_model.get_place_by_tensor_name(name)
+                if tensor:
+                    return tensor
+        node = input_model.get_place_by_operation_name(node_name)
+        if node:
+            tensor = node.get_target_tensor(output_port_index=0)
+            if tensor:
+                return tensor
+        return None
 
+    found_tensor = try_get_tensor(node_name)
+    if found_tensor:
+        found_node_names.append('Tensor:' + found_tensor.get_names()[0])
+        found_nodes.append(found_tensor)
 
-    def try_get_node(model, name, framework):
+    def try_get_node(model, name, framework, node_pre):
```

Comment:
Can we avoid passing this `node_pre` bool? Maybe enum: `operation_consuming`/`operation_producing` or even string

Reply:
added enum